Update excludedUtxos on every account update in send form

komret opened the pull request:
## Description

Previously, `excludedUtxos` were only set during initial render of send form within `getStateFromProps`. However, if a UTXO was added while the send form was open (i.e. when `account` updates), `excludedUtxos` did not update. Coin control is dependant on `excludedUtxos` to organize UTXOs into categories and a UTXO missing from `excludedUtxos` always falls into the default category, thus causing the bug reported in #6906. This could have also lead to bugs in `sendFormBitcoinActions.composeTransaction`, which also relies on `excludedUtxos`, but those have not been reported.

To fix it, I added a `getExcludedUtxos` util and `state.excludedUtxos` is now updated any time `state.account` updates.

I find it weird to keep `excludedUtxos` in send form `state`, but I kept this behaviour as it was designed by @szymonlesisz. I assume it was implemented this way because `composeTransaction` takes `state` as a parameter and we want to avoid extra parameters. Same could be said about other properties of `state`.

## Related Issue

Resolve #6906

matejkriz commented on packages/suite/src/hooks/wallet/useSendFormCompose.ts:
```diff
@@ -273,8 +278,21 @@ export const useSendFormCompose = ({
-    }, [state.account, state.isDirty, account, clearErrors, errors, updateContext]);
+        updateContext({
+            account,
+            excludedUtxos: getExcludedUtxos(account, state.feeInfo, targetAnonymity),
```
this is part of the reason why I always prefer passing more than one param to function in object so it's named (i.e  `getExcludedUtxos({ account, dustLimit: state.feeInfo.dustLimit, targetAnonymity})`)
```suggestion
            excludedUtxos: getExcludedUtxos(account, state.feeInfo.dustLimit, targetAnonymity),
```

komret replied:
Fixed in 8af313bf6e27e0954ccf9ee954ee5f68b7808483. I did not change the params yet to be in line with other utils, but good point and let's discuss that on code improvements

matejkriz reviewed (changes requested):
Except [the one "typo"](https://github.com/trezor/trezor-suite/pull/6929#discussion_r1025324022) makes sense and looks good to me.

Great explanation in description, thanks! 🎖️

szymonlesisz commented:
as far i remember i've "designed it" in personal consultation with you as you needed this data in "coin control" and i needed this info in "compose" and i believe you had no objections then :)

If you are not happy about it lets change it to solution which fits your needs.
Since you already created util we can use it in `sendFormAction` and coin control component separately without keeping it in sendForm state but then it could happen that component and action will be desyncronized.
example: 
component will not render properly for some reason (like the one you are fixing right now) and is displaying "old dataset" while action will have an access to updated state (from getState)

szymonlesisz commented on suite-common/wallet-utils/src/accountUtils.ts:
```diff
@@ -837,3 +838,33 @@ export const readUtxoOutpoint = (outpoint: string) => {
     return { txid: txid.toString('hex'), vout };
 };
+
+export const getExcludedUtxos = (
```
i think it should be placed in sendFormUtils

komret replied:
Moved in 8af313bf6e27e0954ccf9ee954ee5f68b7808483.

komret commented:
I rebased on develop and added a test for the util in 8af313bf6e27e0954ccf9ee954ee5f68b7808483.

@szymon, you are right in both points :) I am keeping the logic as is, although there might be room for improvement here.
